gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxShellTable.java

```diff
@@ -654,5 +655,146 @@ public KnoxShellTable regex(String regex) {
       }
       return table;
     }
+
+    public KnoxShellTable filter(Predicate p) {
+        KnoxShellTable table = new KnoxShellTable();
+        table.headers.addAll(headers);
+        for (List<String> row : rows) {
+          if (p.test(row.get(index))) {
+            table.row();
+            row.forEach(value -> {
+              table.value(value);
+            });
+          }
+        }
+        return table;
+      }
+
+      public KnoxShellTable greaterThan(String comparator) {
```

Comment:
I'm not sure you need multiple copies of each method? `compareTo` should be on the Java `Object` class so should be able to compare any object together without trying to figure out the type.

Reply:
You may consider changing the method signature to `greaterThan(Comparable combarable), lessThan(Comparable comparable),...` to achieve what @risdenk suggested.

Reply:
I don't think this can be done due to the type-specific conversion occurring in each predicate, where the data point from the table is of type String being converted into the method-specific argument type. I might be missing something here, let me know if I am.